src/util/time.h

```diff
@@ -14,18 +14,29 @@
 
 using namespace std::chrono_literals;
 
+/** Mockable clock */
+struct Clock : std::chrono::system_clock {
+    using time_point = std::chrono::time_point<Clock>;
+    /** Return current system time or mocked time, if set */
+    static time_point now() noexcept;
+    static std::time_t to_time_t(const time_point&) = delete; // unused
+    static time_point from_time_t(std::time_t) = delete;      // unused
+};
+using TimeSeconds = std::chrono::time_point<Clock, std::chrono::seconds>;
```

Comment:
I don't think there is a need to. A non-mockable clock is the system clock: `std::chrono::system_clock` (C++11) and `std::chrono::sys_seconds` (C++20). A monotonic clock is `std::chrono::steady_clock`.

Reply:
Well, there's certainly no "need" to -- you could call it `HumptyDumpty` and `WallSeconds` and it would still work...

But the C++ clocks all have an adjective describing their main point (there's also gps_clock and utc_clock and high_resolution_clock and tai_clock and file_clock). It seems like a sensible convention to follow; having `GetTime()` be mockable and `GetTimeSeconds()` be non-mockable has certainly confused me previously...

Reply:
> certainly confused me previously...

Me too, but I think the solution to that is to rename the function names around system time to mention "system" or "sys", similar to what is done in `std::chrono::` itself. Maybe even replace all of them with `std::chrono::system_clock`?

I think there are almost no use cases for a system clock in our codebase. Debug logging might be the only use case, so forcing all other places that deal with time to specify `MockableTimeSeconds` seems verbosity for no benefit. If the only use case of the system clock is for debug logging, which is likely disabled anyway, and one accidentally uses a mockable clock, no user will notice a difference. The only difference could happen in tests, but even there, most tests don't rely on mocktime, so I am highly dubious of any benefits.

Reply:
Could call it by some other adjective; I thought "mock" was the obvious one, since that's how we distinguish the clocks when talking about them anyway. That our clock can stop for extended periods makes it pretty different to regular clocks and even though that's test only behaviour, it's something you need to know about when using it. `NodeClock` maybe? *shrug*

Reply:
Thanks, changed to `NodeClock` in https://github.com/bitcoin/bitcoin/pull/25101